drivers/ad5770r/ad5770r.h

```diff
@@ -0,0 +1,377 @@
+/* AD5770R_OUTPUT_FILTER_CH */
+#define AD5770R_OUTPUT_FILTER_CH_MSK				GENMASK(3, 0)
+#define AD5770R_OUTPUT_FILTER_CH(x)				(((x) & 0xF) << 0)
+
+/* AD5770R_MONITOR_SETUP */
+#define AD5770R_MONITOR_SETUP_MON_FUNCTION_MSK			GENMASK(7, 6)
+#define AD5770R_MONITOR_SETUP_MON_FUNCTION(x)			(((x) & 0x3) << 6)
+#define AD5770R_MONITOR_SETUP_MUX_BUFFER_MSK			BIT(5)
+#define AD5770R_MONITOR_SETUP_MUX_BUFFER(x)			(((x) & 0x1) << 5)
+#define AD5770R_MONITOR_SETUP_IB_EXT_EN_MSK			BIT(4)
+#define AD5770R_MONITOR_SETUP_IB_EXT_EN(x)			(((x) & 0x1) << 4)
+#define AD5770R_MONITOR_SETUP_MON_CH_MSK			GENMASK(3, 0)
+#define AD5770R_MONITOR_SETUP_MON_CH(x)				(((x) & 0x7) << 0)
+
+/* AD5770R_STATUS */
+#define AD5770R_STATUS_BACKGROUND_CRC_STATUS_MSK		BIT(7)
+#define AD5770R_STATUS_IREF_FAULT_MSK				BIT(3)
+#define AD5770R_STATUS_NEGATIVE_CHANNEL0_MSK			BIT(2)
+#define AD5770R_STATUS_OVER_TEMP_MSK				BIT(1)
+#define AD5770R_STATUS_TEMP_WARNING_MSK				BIT(0)
+
+/* AD5770R_HW_LDAC */
+#define AD5770R_HW_LDAC_MASK_CH(x, channel)			(((x) & 0x1) << (channel))
+
+/* AD5770R_CH_DAC */
+#define AD5770R_CH_DAC_DATA_LSB(x)				(((x) & 0x3F) << 2)
+#define AD5770R_CH_DAC_DATA_MSB(x)				(((x) & 0xFF) << 0)
+
+/* AD5770R_CH_SELECT */
+#define AD5770R_CH_SELECT_SEL_CH(x, channel)			(((x) & 0x1) << (channel))
+
+/* AD5770R_CH_INPUT */
+#define AD5770R_CH_DAC_INPUT_DATA_LSB(x)			(((x) & 0x3F) << 2)
+#define AD5770R_CH_DAC_INPUT_DATA_MSB(x)			(((x) & 0xFF) << 0)
+
+/* AD5770R_SW_LDAC */
+#define AD5770R_SW_LDAC_CH(x, channel)				(((x) & 0x1) << (channel))
+
+/* AD5770R_CH_ENABLE */
+#define AD5770R_CH_ENABLE_SET(x, channel)			(((x) & 0x1) << (channel))
+
+#define AD5770R_REG_READ(x)					(((x) & 0x7F) | 0x80)
+#define AD5770R_REG_WRITE(x)					((x) & 0x7F)
+
+enum ad5770r_output_filter_resistor {
+	AD5770R_OUTPUT_FILTER_RESISTOR_60_OHM = 0x0,
+	AD5770R_OUTPUT_FILTER_RESISTOR_5_6_KOHM = 0x5,
+	AD5770R_OUTPUT_FILTER_RESISTOR_11_2_KOHM,
+	AD5770R_OUTPUT_FILTER_RESISTOR_22_2_KOHM,
+	AD5770R_OUTPUT_FILTER_RESISTOR_44_4_KOHM,
+	AD5770R_OUTPUT_FILTER_RESISTOR_104_KOHM,
+};
+
+enum ad5770r_monitor_function {
+	AD5770R_DISABLE = 0,
+	AD5770R_VOLTAGE_MONITORING,
+	AD5770R_CURRENT_MONITORING,
+	AD5770R_TEMPERATURE_MONITORING
+};
+
+enum ad5770r_channels {
+	AD5770R_CH0 = 0,
+	AD5770R_CH1,
+	AD5770R_CH2,
+	AD5770R_CH3,
+	AD5770R_CH4,
+	AD5770R_CH5
+};
+
+enum ad5770r_reference_voltage {
+	AD5770R_EXT_REF_2_5_V = 0,
+	AD5770R_INT_REF_1_25_V_OUT_ON,
+	AD5770R_EXT_REF_1_25_V,
+	AD5770R_INT_REF_1_25_V_OUT_OFF
+};
+
+struct monitor_setup {
+	enum ad5770r_monitor_function		monitor_function;
+	bool 					mux_buffer;
+	bool					ib_ext_en;
+	enum ad5770r_channels			monitor_channel;
+};
+
+struct dac_page_mask {
+	uint16_t 		dac_data_page_mask;
+	uint16_t		input_page_mask;
+};
+
+struct output_range {
+	uint8_t		output_scale;
+	uint8_t		output_range_mode;
+};
+
+struct device_spi_settings {
+	bool			addr_ascension;
+	bool			single_instruction; // for multibyte read/write
+	uint8_t		stream_mode_length;
+};
```

Comment:
What about those structs above, should they be prefixed to avoid namespace issues?